addons/l10n_uy/models/account_move.py

```diff
@@ -22,3 +32,19 @@ def _get_last_sequence_domain(self, relaxed=False):
             where_string += " AND l10n_latam_document_type_id = %(l10n_latam_document_type_id)s"
             param['l10n_latam_document_type_id'] = self.l10n_latam_document_type_id.id or 0
         return where_string, param
+
+    def _get_l10n_latam_documents_domain(self):
+        """ If this is a reversal or debit, suggest only related subtypes """
+        self.ensure_one()
+        domain = super()._get_l10n_latam_documents_domain()
+        if self.journal_id.country_code == "UY":
+            # We do this only to be able to match with the subtypes
+            if original_document := self.reversed_entry_id or self.debit_origin_id:
+                codes = self.env["l10n_latam.document.type"].search(domain).mapped('code')
+                for sublist_type in UY_DOC_SUBTYPES:
+                    if original_document.l10n_latam_document_type_id.code in sublist_type:
+                        break
+                codes = list(set(codes).intersection(sublist_type))
+                if codes:
+                    domain += [("code", "in", codes)]
```

Comment:
I believe it's better like that
```suggestion
        if self.country_code == "UY" and (original_move := self.reversed_entry_id or self.debit_origin_id):
            matching_subtype_codes = [
                subtype for subtype in UY_DOC_SUBTYPES
                if original_move.l10n_latam_document_type_id.code in subtype
            ]
            if matching_subtype_codes:
                # restrict to the codes from the subtype matching the one of the original_move (e.g. 'e-ticket')
                codes = self.env["l10n_latam.document.type"].search(domain).mapped('code')
                allowed_codes = set(codes).intersection(set(matching_subtype_codes[0]))
                domain += [("code", "in", tuple(allowed_codes))]